packages/webpack/web-webpack-plugin/test/cases/main-thread/page-config/rspack.config.js

```diff
@@ -0,0 +1,64 @@
+import { LynxTemplatePlugin } from '@lynx-js/template-webpack-plugin';
+import { WebWebpackPlugin } from '../../../../src';
+
+/** @type {import('@rspack/core').Configuration} */
+export default {
+  entry: {
+    a: './a.js',
+    b: './b.js',
+    'a:main-thread': { import: './a.js', filename: 'a/a.lepus.js' },
+    'b:main-thread': { import: './b.js', filename: 'b/b.lepus.js' },
+  },
+  context: __dirname,
+  output: {
+    filename: '[name]/[name].js',
+  },
+  plugins: [
+    new WebWebpackPlugin({
+      include: [/a.js/g],
+    }),
+    new LynxTemplatePlugin({
+      ...LynxTemplatePlugin.defaultOptions,
+      chunks: ['a', 'a:main-thread'],
+      filename: 'a/template.js',
+      intermediate: '.rspeedy/a',
+    }),
+
+    compiler => {
+      compiler.hooks.thisCompilation.tap('test', (compilation) => {
+        compilation.hooks.processAssets.tap('test', () => {
+          ['a'].forEach(name => {
+            const asset = compilation.getAsset(`${name}/${name}.js`);
+            compilation.updateAsset(asset.name, asset.source, {
+              ...asset.info,
+              'lynx:main-thread': true,
+            });
+          });
+        });
+
+        compilation.hooks.processAssets.tap(
+          {
+            name: 'page-config-test',
+            stage:
+              compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_PRE_PROCESS,
+          },
+          () => {
+            const hooks = LynxTemplatePlugin.getLynxTemplatePluginHooks(
+              // @ts-expect-error compilation is compatible with webpack
+              compilation,
+            );
+            hooks.beforeEncode.tap('page-config-test', (args) => {
```

Comment:
This test doesn't seem to validate functionality effectively; it only verifies the implementation.  

Could you add an end-to-end (e2e) test to ensure that configurations can be properly read at runtime?

Reply:
I don't think this is needed in this PR, in this PR we are only concerned with the final outputs, as it only involves build time.

Of course, it makes sense to have this test at runtime, but it should not appear in this PR